### PR TITLE
pre-reqs for sonar pipeline integration

### PR DESCRIPTION
**Pull request by ICoxCH**
added npm scripts to run jes tests with coverage, and added sonar properties to allow publishing report to sonar

### CODE REVIEW

File: jest.config.coverage.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
```
Comment (mwestacott) [inline jest.config.coverage.js]: As discussed, not sure we need this file as we have a similar one already pushed.

File: package.json [hunk 2]
```diff
@@ -7,7 +7,9 @@
     "build": "tsc && cp -r views dist/",
     "devstart": "nodemon",
     "test": "jest --forceExit",
-    "lint": "tslint -p ."
+    "lint": "tslint -p .",
+    "sonarqube": "npm run coverage-jest && sonar-scanner",
+    "coverage-jest": "jest -c jest.config.coverage.js --forceExit --coverage"
```
Comment (mwestacott) [inline package.json]: Target could be just 'coverage', for consistency with the 'test' target above, which also uses Jest but doesn't mention it in the target name.

File: package.json [hunk 3]
```diff
@@ -7,7 +7,9 @@
     "build": "tsc && cp -r views dist/",
     "devstart": "nodemon",
     "test": "jest --forceExit",
-    "lint": "tslint -p ."
+    "lint": "tslint -p .",
+    "sonarqube": "npm run coverage-jest && sonar-scanner",
+    "coverage-jest": "jest -c --forceExit --coverage"
```
Comment (mwestacott) [inline package.json]: See previous comment about the target name.

Also, just curious as to what the -c flag does?

Comment (ICoxCH) [reply package.json]: -c is if you want to define a non default config (I should of removed this when removing jest.config.coverage.ts)

**Pull request merged**